File: src/main/java/tax/Vehicle.java
package tax;

import java.time.LocalDate;

class Vehicle {
    private final int co2Emissions;
    private final FuelType fuelType;
    private LocalDate dateOfFirstRegistration;
    private int listPrice;

    Vehicle(int co2Emissions, FuelType fuelType, LocalDate dateOfFirstRegistration, int listPrice) {
        this.co2Emissions = co2Emissions;
        this.fuelType = fuelType;
        this.dateOfFirstRegistration = dateOfFirstRegistration;
        this.listPrice = listPrice;
    }

    public int getCo2Emissions() {
        return co2Emissions;
    }

    public FuelType getFuelType() {
        return fuelType;
    }

    public LocalDate getDateOfFirstRegistration() {
        return dateOfFirstRegistration;
    }

    public void setDateOfFirstRegistration(LocalDate dateOfFirstRegistration) {
        this.dateOfFirstRegistration = dateOfFirstRegistration;
    }
}

File: exercises/dotnet/TaxCalculator/Vehicle.cs
using System;

namespace TaxCalculator
{
    public class Vehicle
    {
        public int Co2Emissions { get; }
        public FuelType FuelType { get; }
        public DateTime DateOfFirstRegistration { get; }
        public int ListPrice { get; }
        public bool IsRde2Compliant { get; }

        public Vehicle(int co2Emissions, FuelType fuelType, DateTime dateOfFirstRegistration, int listPrice, bool isRde2Compliant = true )
        {
            Co2Emissions = co2Emissions;
            FuelType = fuelType;
            DateOfFirstRegistration = dateOfFirstRegistration;
            ListPrice = listPrice;
            IsRde2Compliant = isRde2Compliant;
        }
    }
}
